pages/impact/[address].tsx

```diff
@@ -120,14 +152,62 @@ export default function Impact({ balance }: Props) {
 export async function getServerSideProps({
   params,
 }: GetServerSidePropsContext) {
-  const { chain } = getNetwork();
-  const address = params?.address;
-  const balance = await getBalance(address as string, chain?.id);
-  const decimals = BigInt(1000000000000000000);
-  const formattedBalance = balance.div(decimals).toNumber();
+  const address = params?.address as string;
+  const chains = isProd() ? supportedChains.mainnet : supportedChains.testnet;
+
+  const valueFromKv = (await kv.get(address)) as string;
+  if (valueFromKv) {
+    const dateFirstGlo: Date = new Date(valueFromKv);
+    return {
+      props: {
+        whenFirstGlo: beautifyDate(dateFirstGlo),
+      },
+    };
+  }
+
+  const transactions: { [id: number]: Erc20Transaction | null } = {};
+  for (const chain of chains) {
+    const chainHex = `0x${numberToHex(chain)}`;
+    transactions[chain] = await fetchFirstGloTransaction(address, chainHex);
+  }
+
+  const timeStamps = Object.values(transactions).map(
+    (tx) => tx?.blockTimestamp
+  );
+
+  const firstGlo = getEarliest(timeStamps);
```

Comment:
Can't you sort by timestamp while fetching from Moralis?

Reply:
Can't do cross chain afaik?